components/common/BraveCoreThemeProvider.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react'
+import { ThemeProvider } from 'brave-ui/theme'
+import IBraveTheme from 'brave-ui/theme/theme-interface'
+
+export type BraveCoreThemeProviderProps = {
+  initialThemeType?: chrome.braveTheme.ThemeType
+  dark: IBraveTheme,
+  light: IBraveTheme
+}
+type BraveCoreThemeProviderState = {
+  braveCoreTheme?: chrome.braveTheme.ThemeType
+}
+
+export default class BraveCoreThemeProvider
+
+extends React.Component<BraveCoreThemeProviderProps, BraveCoreThemeProviderState> {
+  componentWillMount () {
```

Comment:
could `componentDidMount` do the trick here since it would re-render children anyway? if you think this is invalid maybe explicit tell this method deprecation using `UNSAFE_componentWillMount`?

Reply:
Good catch. I'll use the constructor since `componentDidMount` will cause an extra render